libs/utils/utils.js

```diff
@@ -1391,38 +1391,82 @@ export function enablePersonalizationV2() {
   const enablePersV2 = getMepEnablement('personalization-v2');
   return !!enablePersV2 && isSignedOut();
 }
+export function getCookie(key) {
+  const cookie = document.cookie.split(';')
+    .map((x) => decodeURIComponent(x.trim()).split(/=(.*)/s))
+    .find(([k]) => k === key);
+  return cookie ? cookie[1] : null;
+}
+export function normCountry(country) {
+  if (!country) return null;
+  let c = country?.toLowerCase();
+  if (c === 'uk') c = 'gb';
+  return c.split('_')[0];
+}
+export async function getC0002(optin, akamaiLocale) {
```

Comment:
The method is a bit confusing: it returns a country as well as whether consent is provided. We should rewrite like
```
const country = await getCountry(); // promise that returns value from session storage or tries to fetch from geo2 API.
if (country !== 'gb') return;
const hasConsentedForC0002 = hasProvidedConsent() && getConsent().includes('C0002); //has provided consent -> `OptanonAlertBoxClosed` is defined and it is a proper date, `getConsent` -> returns the `groups=` value from `OptanonConsent` cookie and C0002:1 is there.